index.js

```diff
@@ -1 +1,367 @@
+
+    blockHeader.appendChild(blockTitle)
+    blockHeader.appendChild(blockBtn)
+
+    listBlock.appendChild(blockHeader)
+
+    const blockBody = document.createElement('div');
+    blockBody.classList.add('block-body')
+    blockBody.classList.add('hidden')
+
+    blockBtn.addEventListener('click', () => blockBody.classList.toggle('hidden'))
+
+    listBlock.appendChild(blockBody)
+
+    contactListEl.appendChild(listBlock)
+
+    value.forEach((item) => {
+      renderContactCard(item, blockBody)
+    })
+  })
+}
+
+const isElementEmpty = (element) => {
+  return element.children.length === 0;
+}
+
+function clearInputs(inputs) {
+  inputs.forEach((inp) => inp.value = '')
+}
+
+function createContactItemFromInputs(inputs) {
+  const contactItem = {}
+  inputs.map((inp) => contactItem[inp.dataset.content] = inp.value)
+
+  return contactItem
+}
+
+function addContact(inputs) {
+  const contactItem = createContactItemFromInputs(inputs)
+
+  const firstChar = contactItem.name.charAt(0).toLowerCase()
+
+  clearInputs(inputs)
+
+  let contacts = JSON.parse(localStorage.getItem('contacts')) || {};
+
+  if (!contacts[firstChar]) {
+    contacts[firstChar] = [];
+  }
+
+  contacts[firstChar].push(contactItem);
+  localStorage.setItem('contacts', JSON.stringify(contacts));
+
+  renderList()
+}
+
+function clearList() {
+  localStorage.removeItem('contacts');
+  renderEmptyList()
+}
+
+function handleContact(targetInputs, submitEl) {
+  let handleStatus;
+  const contactItem = createContactItemFromInputs(targetInputs)
+
+  if (!isContactNew(contactItem)) { 
+    showElementsError(submitEl, 'You already have this contact');
+    return
+  }
+
+  const validationErrors = [];
+
+  targetInputs.forEach((inp) => {
+    const validationRes = validateValue(inp.value, validationRules[inp.dataset.content])
+
+    validationRes && validationErrors.push({ invalidInput: inp, errorMessage: validationRes})
+  })
+
+  if (validationErrors.every((x) => x.errorMessage === null)) {
+    addContact(targetInputs);
+    handleStatus = true;
+  } else {
+    validationErrors.map((x) => showElementsError(x.invalidInput, x.errorMessage))
+    handleStatus = false;
+  }
+
+  return handleStatus;
+}
+
+function decreaseTitleCounter(titleEl) {
+  const counter = titleEl.textContent.match(/\d/);
+
+  titleEl.textContent = titleEl.textContent.replace(/\d/, counter - 1)
+}
+
+function disableScroll() {
+  document.body.classList.add('no-scroll');
+}
+
+function allowScroll() {
+  document.body.classList.remove('no-scroll');
+}
+
+function getContactListItems() {
+  const contactList = JSON.parse(localStorage.getItem('contacts')) || {};
+  return Object.values(contactList).flat()
+}
+
+function isContactNew(contactItem) {
+  return !getContactListItems().find((savedContact) => {
+    return savedContact.name === contactItem.name && 
+    savedContact.vacancy === contactItem.vacancy &&
+    savedContact.phone === contactItem.phone
+  })
+}
```

Comment:
Тебе приходится делать !, чтоб привести к boolean. Можно облегчить жизнь:

```suggestion
function isNewContact(contactItem) {
const contacts = getContactListItems();
return contacts.every((contact) => contact.name !== contactItem.name && ... &&)
}
```